src/libraries/Common/src/System/Security/Cryptography/Asn1Reader/AsnWriter.cs

```diff
@@ -11,16 +11,17 @@
 using System.Diagnostics;
 using System.Runtime.InteropServices;
 
+#nullable enable
 namespace System.Security.Cryptography.Asn1
 {
     /// <summary>
     ///   A writer for BER-, CER-, and DER-encoded ASN.1 data.
     /// </summary>
     internal sealed partial class AsnWriter : IDisposable
     {
-        private byte[] _buffer;
+        private byte[]? _buffer;
```

Comment:
I'd just call this non-nullable and get rid of all the bangs and asserts.

Reply:
yes it is causing lot of `!`, but it is not inti-ed in constructor, only init-ed in EnsureWriteCapacity(), which case we normally make it nullable so that in the future devs were aware of that it could be null. Though I am ok with making it non null if this API is not expecting much maintenance.

Reply:
Yeah, let's go with non-null.  (Not calling EnsureWriteCapacity would cause an out of bounds write, so by the time anything is writing it's also not null)

Reply:
OK!